chainer/functions/pooling/average_pooling_nd.py

```diff
@@ -0,0 +1,148 @@
+import numpy
+
+import functools
+import operator
+import six
+
+from chainer import cuda
+from chainer.functions.pooling import average_pooling_nd_kernel
+from chainer.functions.pooling import pooling_nd
+from chainer import utils
+from chainer.utils import conv_nd
+
+
+if cuda.cudnn_enabled:
+    cudnn = cuda.cudnn
+    libcudnn = cudnn.cudnn
+    _cudnn_version = libcudnn.getVersion()
+
+
+class AveragePoolingND(pooling_nd._PoolingND):
+
+    """Average pooling over a set of N-dimensional planes."""
+    # TODO(takagi) Support cover_all mode.
+
+    def __init__(self, ndim, ksize, stride=None, pad=0, use_cudnn=True):
```

Comment:
Should we add `cover_all` argument for later use? We can add new argument only in the last to keep backward compatibility, but if we add `cover_all` argument in the tail, the order of arguments are inconsistent with that of  `MaxPoolingND`.